docs/usage.rst

```diff
@@ -0,0 +1,719 @@
+Usage
+=====
+
+The aim of
+:mod:`audbackend`
+is to provide an
+abstract interface for
+any kind of file storage system.
+Even those,
+that have not been
+invented yet :)
+
+This tutorial is divided
+into two parts.
+In the first part,
+we show the basic usage
+by means of a
+standard file system.
+In the second part,
+we take a deep dive
+and develop a backend
+that stores files into
+a SQLite_ database.
+
+.. set temporal working directory
+.. jupyter-execute::
+    :hide-code:
+
+    import os
+    import audeer
+
+    _cwd_root = os.getcwd()
+    _tmp_root = audeer.mkdir(os.path.join('docs', 'tmp'))
+    os.chdir(_tmp_root)
+
+.. helper functions
+.. jupyter-execute::
+    :hide-code:
+
+    import functools
+
+    def add_method(cls):
+        def decorator(func):
+            @functools.wraps(func)
+            def wrapper(self, *args, **kwargs):
+                return func(self, *args, **kwargs)
+            setattr(cls, func.__name__, wrapper)
+            return func
+        return decorator
+
+
+Basics
+------
+
+The heart of
+:mod:`audbackend`
+is the class
+:class:`audbackend.Backend`,
+which provides an abstract
+interface to communicate
+with a storage system.
+The class
+:class:`audbackend.FileSystem`
+implements
+:class:`audbackend.Backend`
+for a standard file system.
+
+
+We start by registering the class
+(in fact,
+the class is registered
+by default,
+but it doesn't hurt
+if we do it again).
+
+.. jupyter-execute::
+
+    import audbackend
+
+    audbackend.register('file-system', audbackend.FileSystem)
+
+
+We create an instance
+and provide two arguments:
+
+* ``host``: the host address,
+  in this case a folder on the local file system.
+* ``repository``: the repository name,
+  in this case a sub-folder within the ``host`` folder
+  (it is possible to have several repositories
+  on the same host).
+
+.. jupyter-execute::
+
+    backend = audbackend.create('file-system', './host', 'repo')
+
+
+This will create an empty repository
+(in our case the folder ``'./<host>/<repo>/'``).
+To access an existing repository,
+we would do:
+
+.. jupyter-execute::
+
+    backend = audbackend.access('file-system', './host', 'repo')
+
+
+To put a file on the backend,
+we provide two path arguments.
+
+* ``src_path``: path to a file on the local file system.
+  This is the file we want to store on the backend.
+* ``dst_path``: virtual path that represents the file on the backend.
+  It serves as an alias that is understood by all backends.
```

Comment:
At some point we need to add that the path should always start with `/`. Maybe here is the right location?